roomserver/storage/tables/interface.go

```diff
@@ -173,6 +174,20 @@ type Redactions interface {
 	MarkRedactionValidated(ctx context.Context, txn *sql.Tx, redactionEventID string, validated bool) error
 }
 
+type Purge interface {
+	PurgeEventJSONs(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeEvents(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeRoom(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeStateSnapshots(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeStateBlocks(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgePreviousEvents(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeInvites(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeMemberships(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgePublished(ctx context.Context, txn *sql.Tx, roomID string) error
+	PurgeRedactions(ctx context.Context, txn *sql.Tx, roomNID types.RoomNID) error
+	PurgeRoomAliases(ctx context.Context, txn *sql.Tx, roomID string) error
```

Comment:
As previously mentioned, this seems crazy to have 11 functions which need to be called sequentially else it isn't purging the room. Just do it all in a single function?!